onnxruntime/core/providers/cuda/tensor/gather_elements_impl.cu

```diff
@@ -3,7 +3,7 @@
 
 #include "core/providers/cuda/tensor/gather_elements_impl.h"
 #include "core/providers/cuda/tensor/scatter_elements_impl.h"
-#ifdef ENABLE_TRAINING
+#ifdef ENABLE_TRAINING_OPS
```

Comment:
So this PR is trying to make all training ops related with ENABLE_TRAINING_OPS. And ENABLE_TRAINING_OPS=True when either ENABLE_TRAINING or ENABLE_TRAINING_ON_DEVICE is enabled?

Reply:
Right!